spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/service/connection/ConnectionDetailsFactories.java

```diff
@@ -122,13 +122,29 @@ record Registration<S, D extends ConnectionDetails>(Class<S> sourceType, Class<D
 		@SuppressWarnings("unchecked")
 		private static <S, D extends ConnectionDetails> Registration<S, D> get(ConnectionDetailsFactory<S, D> factory) {
 			ResolvableType type = ResolvableType.forClass(ConnectionDetailsFactory.class, factory.getClass());
-			if (!type.hasUnresolvableGenerics()) {
+			if (isResolved(type)) {
```

Comment:
Of course, I've just noticed after submitting that `type.resolveGenerics()` is called twice now. Let me know if you want me to refactor.